src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -147,52 +148,38 @@ public virtual async Task<int> SendJsonRpcResult(JsonRpcResult result)
                     await enumerator.DisposeAsync();
                 }
 
-                await _handler.SendRawAsync(_jsonClosingBracket, true);
-                singleResponseSize += 1;
-
-                return singleResponseSize;
+                await resultData.WriteAsync(_jsonClosingBracket);
             }
             else
             {
-                return await SendJsonRpcResultEntry(result.SingleResponse!.Value);
+                SendJsonRpcResultEntry(resultData, result.SingleResponse!.Value);
             }
+
+            // ? What if we write more than int.MaxValue.
+            // Result could be negative
+            // return (int)resultData.WrittenBytes;
+            return (int)resultData.WrittenBytes;
```

Comment:
Technically, we could send infinite number of bytes, but our result is bounded by `int.MaxValue` according to the result type of this operation. 

The `CounterStream` uses a long internally, but this still could be problematic (we are bounded by `long.MaxSize`).